test-data/unit/check-parameter-specification.test

```diff
@@ -1112,3 +1112,18 @@ def func(callback: Callable[P, str]) -> Callable[P, str]:
         return "foo"
     return inner
 [builtins fixtures/paramspec.pyi]
+
+[case testParamSpecArgsAndKwargsMissmatch]
+from typing import Callable
+from typing_extensions import ParamSpec
+
+P = ParamSpec("P")
+
+def func(callback: Callable[P, str]) -> Callable[P, str]:
+    def inner(
+        *args: P.kwargs,   # E: Use ".args" for "*" argument
+        **kwargs: P.args,  # E: Use ".kwargs" for "**" argument
+    ) -> str:
+        return "foo"
```

Comment:
Maybe add some `reveal_type`s here, as a sanity check to make sure that the typo doesn't cause mypy to make any incorrect inferences elsewhere?

```suggestion
        reveal_type(args)  # N: Revealed type is "builtins.tuple[builtins.object, ...]"
        reveal_type(kwargs)  # N: Revealed type is "builtins.dict[builtins.str, builtins.object]"
        return "foo"
```

Reply:
This is not true:

```
main:11: note: Revealed type is "P1.args`-1"  (diff)
main:12: note: Revealed type is "P1.kwargs`-1" (diff)
```

Reply:
> This is not true:
> 
> ```
> main:11: note: Revealed type is "P1.args`-1"  (diff)
> main:12: note: Revealed type is "P1.kwargs`-1" (diff)
> ```

Oh, right. Maybe this, in that case?

```suggestion
        a = args[0]
        reveal_type(a)  # N: Revealed type is "builtins.object"
        for b, c in kwargs.items():
            reveal_type(b)  # N: Revealed type is "builtins.str"
            reveal_type(c)  # N: Revealed type is "builtins.object"
        return "foo"
```

Reply:
I think this is too much :)

Reply:
Fair enough :)